espresso-support/core/src/main/java/com/novoda/espresso/AccessibilityViewMatchers.java

```diff
@@ -0,0 +1,187 @@
+package com.novoda.espresso;
+
+import android.content.res.Resources;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringRes;
+import android.view.View;
+import android.view.accessibility.AccessibilityNodeInfo;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import static org.hamcrest.Matchers.is;
+
+public class AccessibilityViewMatchers {
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnClick(@StringRes final int resourceId) {
+        return new TypeSafeMatcher<View>() {
+
+            private String resourceName = null;
+            private String expectedText = null;
+
+            @Override
+            protected boolean matchesSafely(View view) {
+                if (!view.isClickable()) {
+                    return false;
+                }
+
+                if (null == this.expectedText) {
+                    try {
+                        expectedText = view.getResources().getString(resourceId);
+                        resourceName = view.getResources().getResourceEntryName(resourceId);
+                    } catch (Resources.NotFoundException ignored) {
+                        // view could be from a context unaware of the resource id.
```

Comment:
I can't see how it could ever be the case, but it means that the `context` we use here might not be aware of the resources set that contains the string resource. Like if you use application context to read an attribute that's only on the activity theme.

that's what the comment refers too. The resourceNotFound is legit though - user might pass something willy nilly.

Just added it as it's like this in the existing ones.